resources/assets/components/CampaignInbox/index.js

```diff
@@ -61,4 +62,31 @@ class CampaignInbox extends React.Component {
   }
 }
 
+CampaignInbox.propTypes = {
+  campaign: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
```

Comment:
This totally isn't necessary for this pull request, but we should try to avoid using `PropTypes.object` where we can, since it doesn't give us any assurances about the contents of the object. One good solution might be to create a `types.js` file somewhere and export `PropType.shape` definitions from there...

```js
// types.js
export const CampaignType = PropTypes.shape({
  id: PropTypes.string.isRequired,
  // ...
});


// CampaignInbox/index.js
import { CampaignType } from '../../../types.js';

CampaignInbox.PropTypes = {
  campaign: CampaignType,
  // ...
};

```

Reply:
oh interesting. I really like that! The only reason Ignored it was because I didn't want to have to shape out all of these large objects since we are doing things like passing in the entire campaign object, user object, etc. But this is a really nice solution to this. I will maybe do that in a separate PR so I can keep the work chunked.